### ISSUE TITLE
Implement the recursive filter

### ISSUE THREAD

**Issue by NigelRoberts** [metoppv/improver#192]
Prerequisite: #137 - Done

As the product owner I want IMPROVER to have the capability to run the recursive filter to convert a square neighbourhood into a Gaussian-like kernel or simply to smooth over short distances.  The recursive filter uses an "alpha" parameter that controls what proportion of stuff (probability) is passed along to the next grid square. 

The recursive filter is currently used in the GPP and has been written in Fortran. It has been tested within Python but not yet written in Python. See link below by Stephen Moseley.
See PostProc/wiki/ticket/1962/NeighbourhoodAlgorithm (recursivefilter.f90).

The implementation envisaged for IMPROVER should allow the alpha parameter to be set on a grid square by grid square basis (e.g. an array of alpha parameters equivalent in dimensions to the domain). Each grid square will actually need a pair of alpha parameters to allow for orthogonally discrete functionality, e.g. one to act in the N-S direction, and another to act in the E-W direction.

Some thought will be required about how to set up the alpha parameters, or whether these should be created elsewhere and passed in for use by the recursive filter. At this point the alpha parameters will be on a grid, but there will be the same alpha value at each gridpoint. There will be the option for different alpha value in the x and y directions.

Acceptance criteria:
 * Unit tests
 * Visual inspection of idealised fields - with particular attention paid to the domain edges
 * Test on real fields and confirm the results are sensible
 * Pay attention to run time, which if slow should be flagged as in need of optimisation.
 * Make results of testing available.
 * Create plugin only, CLI will be covered by another ticket.

**Comment by bayliffe** [comment]
Completed.
